synapseclient/core/download/download_async.py

```diff
@@ -0,0 +1,477 @@
+"""Logic required for the actual transferring of files."""
```

Comment:
`download_async.py` file is close to a copy of `download_threads.py`. No significant changes here, but I wanted to start to point files to the correct locations they're going to eventually live at.